qa/os/src/test/java/org/elasticsearch/packaging/test/ArchiveTests.java

```diff
@@ -107,28 +119,68 @@ public void test32SpecialCharactersInJdkPath() throws Exception {
         }
     }
 
-    public void test50StartAndStop() throws Exception {
+    public void test40AutoconfigurationNotTriggered() throws Exception {
+        ServerUtils.addSettingToExistingConfiguration(installation, "discovery.seed_hosts", "[\"127.0.0.1:9300\"]");
+        startElasticsearch();
+        verifySecurityNotAutoConfigured(installation);
+        stopElasticsearch();
+        ServerUtils.removeSettingFromExistingConfiguration(installation, "discovery.seed_hosts");
+
+        ServerUtils.addSettingToExistingConfiguration(installation, "node.roles", "[\"voting_only\", \"master\"]");
+        startElasticsearch();
+        verifySecurityNotAutoConfigured(installation);
+        stopElasticsearch();
+        ServerUtils.removeSettingFromExistingConfiguration(installation, "node.roles");
+
+        ServerUtils.addSettingToExistingConfiguration(installation, "node.roles", "[\"ingest\"]");
+        startElasticsearch();
+        verifySecurityNotAutoConfigured(installation);
+        stopElasticsearch();
+        ServerUtils.removeSettingFromExistingConfiguration(installation, "node.roles");
+
+        ServerUtils.addSettingToExistingConfiguration(installation, "xpack.security.http.ssl.enabled", "false");
+        startElasticsearch();
+        verifySecurityNotAutoConfigured(installation);
+        stopElasticsearch();
+        ServerUtils.removeSettingFromExistingConfiguration(installation, "xpack.security.http.ssl.enabled");
+    }
+
+    public void test41AutoConfigurationNotTriggeredOnNotWriteableConfDir() throws Exception {
+        Path tempDir = createTempDir("custom-config");
+        Path tempConf = tempDir.resolve("elasticsearch");
+        FileUtils.copyDirectory(installation.config, tempConf);
+        Platforms.onLinux(() -> sh.run("chown -R elasticsearch:elasticsearch " + tempDir));
+        Platforms.onLinux(() -> sh.run("chmod -w " + tempDir));
+        sh.getEnv().put("ES_PATH_CONF", tempConf.toString());
+        startElasticsearch();
+        verifySecurityNotAutoConfigured(installation);
+        stopElasticsearch();
+        sh.getEnv().remove("ES_PATH_CONF");
+        IOUtils.rm(tempDir);
+    }
```

Comment:
These tests look great!
Some of them might be worth running in docker as well, but I'm content with them like this, and probably their maintainers know better :)